src/main/java/nextstep/ladder/view/InputView.java

```diff
@@ -21,19 +19,67 @@
         }
     }
 
-    private static List<UserName> makeUsers() {
+    private static List<User> makeUsers() {
         String names;
         System.out.println("참여할 사람 이름을 입력하세요. (이름은 쉼표(,)로 구분하세요)");
         names = SCANNER.next();
         String[] userNameList = names.split(",");
-        List<UserName> users = new ArrayList<>();
-        for (String name : userNameList) {
-            UserName userName = new UserName(name);
-            users.add(userName);
-        }
+        List<User> users = new ArrayList<>();
+
+        IntStream.range(0, userNameList.length).forEach(index -> users.add(new User(userNameList[index], index)));
+        
         return users;
     }
 
+    public static List<String> getResults() {
+        while (true) {
+            try {
+                return makeResults();
+            } catch (IllegalArgumentException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+    }
+
+    public static void printResults(List<User> users, List<String> results) {
+        while (true) {
+            try {
+                System.out.println("결과를 보고 싶은 사람은?");
+
+                String userName = SCANNER.next();
+                printResultByName(userName, users, results);
+            } catch (IllegalArgumentException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+    }
+
+    private static void printResultByName(String userName, List<User> users, List<String> results) {
+        if ("all".equals(userName)) {
+            System.out.println("실행 결과");
+            users.stream().
+                    forEach(user -> System.out.println(user + " : " + user.getResult(results)));
+            return;
+        }
+
+        Optional<User> matchUser = users.stream().filter(user -> user.toString().equals(userName)).findFirst();
+        User user = matchUser.orElseThrow(() -> new IllegalArgumentException("유저를 찾을 수 없습니다"));
```

Comment:
```suggestion
        User user = users.stream()
        .filter(user -> user.toString().equals(userName))
        .findFirst();
        .orElseThrow(() -> new IllegalArgumentException("유저를 찾을 수 없습니다"));
```

위 와같이 사용해도 동일한 결과를 얻을 수 있어요.

추가로 스트림은 행위 단위로 내려쓰는것이 가독성과 디버깅에 유리한점이 많다고 생각합니다.

Reply:
```JAVA
user -> user.toString().equals(userName)
```
게터를 사용하지 않고 둘의 동등성을 구할 수 있도록 개선하면 좋을 것 같아요.